src/overlays/actors/ovl_En_Zl2/z_en_zl2.c

```diff
@@ -27,265 +101,1664 @@ const ActorInit En_Zl2_InitVars = {
+    f32 posZ;
+
+    if (this->unk_248 == 0) {
+        posX = this->actor.posRot.pos.x;
+        posY = this->actor.posRot.pos.y + (kREG(5) + -26.0f);
+        posZ = this->actor.posRot.pos.z;
+
+        Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0,
+                            0x4000, 0, 3);
+        this->unk_248 = 1;
+    }
+}
+
+void func_80B5008C(EnZl2* this) {
+    Actor* attachedB = this->actor.attachedB;
+
+    if (attachedB != NULL) {
+        attachedB->posRot.pos.x = this->actor.posRot.pos.x;
+        attachedB->posRot.pos.y = this->actor.posRot.pos.y + (kREG(5) + -26.0f);
+        attachedB->posRot.pos.z = this->actor.posRot.pos.z;
+    }
+}
+
+void func_80B500E0(EnZl2* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction = func_80B4ED94(globalCtx, 0);
+    Vec3f* thisPos = &this->actor.posRot.pos;
+    f32 startX;
+    f32 startY;
+    f32 startZ;
+    f32 endX;
+    f32 endY;
+    f32 endZ;
+    f32 someFloat;
+
+    if (npcAction != NULL) {
+        someFloat = func_8006F9BC(npcAction->endFrame, npcAction->startFrame, globalCtx->csCtx.frames, 8, 8);
+        startX = npcAction->startPos.x;
+        startY = npcAction->startPos.y;
+        startZ = npcAction->startPos.z;
+        endX = npcAction->endPos.x;
+        endY = npcAction->endPos.y;
+        endZ = npcAction->endPos.z;
+        thisPos->x = ((endX - startX) * someFloat) + startX;
+        thisPos->y = ((endY - startY) * someFloat) + startY;
+        thisPos->z = ((endZ - startZ) * someFloat) + startZ;
+    }
+}
+
+void func_80B501C4(EnZl2* this, s32 arg1) {
+    if (this->actor.attachedB != NULL) {
+        ((DoorWarp1*)this->actor.attachedB)->unk_1A8 = arg1;
+    }
+}
+
+void func_80B501E8(EnZl2* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction = func_80B4ED94(globalCtx, 0);
+    s32 temp_f16;
+
+    if (npcAction != NULL) {
+        temp_f16 = (1.0f - func_8006F93C(npcAction->endFrame, npcAction->startFrame, globalCtx->csCtx.frames)) * 255.0f;
+        this->unk_1A8 = temp_f16;
+        this->actor.shape.unk_14 = temp_f16;
+        func_80B501C4(this, temp_f16);
+    }
+}
+
+void func_80B50260(EnZl2* this, GlobalContext* globalCtx) {
+    this->action = 1;
+    this->drawConfig = 0;
+    this->actor.shape.unk_14 = 0;
+}
+
+void func_80B50278(EnZl2* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction = func_80B4ED94(globalCtx, 0);
+    s16 actionRotY;
+
+    this->actor.posRot.pos.x = npcAction->startPos.x;
+    this->actor.posRot.pos.y = npcAction->startPos.y;
+    this->actor.posRot.pos.z = npcAction->startPos.z;
+    actionRotY = npcAction->rot.y;
+    this->actor.shape.unk_14 = 0xFF;
+    this->action = 2;
+    this->drawConfig = 1;
+    this->actor.shape.rot.y = actionRotY;
+    this->actor.posRot.rot.y = actionRotY;
+}
+
+void func_80B50304(EnZl2* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    ActorShape* actorShape = &this->actor.shape;
```

Comment:
a temp for shape is kinda weird, i assume its required?

Reply:
I'll double check, but it probably was

Reply:
yeah, it was